src/azure-cli-core/azure/cli/core/aaz/_arg.py

```diff
@@ -562,18 +574,20 @@ def __init__(
             self, help="Name or ID of subscription. You can configure the default subscription "
                        "using `az account set -s NAME_OR_ID`",
             fmt=None,
+            completer=None,
             **kwargs):
+        from azure.cli.core._completers import get_subscription_id_list
         fmt = fmt or AAZSubscriptionIdArgFormat()
+        completer = completer or get_subscription_id_list
         super().__init__(
             help=help,
             fmt=fmt,
+            completer=completer,
             **kwargs
         )
 
     def to_cmd_arg(self, name, **kwargs):
```

Comment:
Can clean up this method overwritten.

```suggestion
    def to_cmd_arg(self, name, **kwargs):
```

Reply:
Done